src/endpoints/phenotypes.ts

```diff
@@ -1,17 +1,101 @@
-import { idKey } from '../env';
-import { ArrangerProject, searchSqon } from '../sqon/searchSqon';
-import { SetSqon } from './sets/setsTypes';
 import { get } from 'lodash';
+
+import { idKey } from '../env';
+import { ArrangerProject, runQuery } from '../sqon/searchSqon';
 import { replaceSetByIds } from '../sqon/setSqon';
+import { SetSqon } from './sets/setsTypes';
 
 const MAX_PHENOTYPES = 100000;
 
+const extractPsIds = (resp): string[] => (resp?.data?.participant?.hits?.edges || []).map(edge => edge.node[idKey]);
+
+const throwErrorsFromGqlQueryIfExist = resp => {
```

Comment:
Seems to be useful for others places, should we put it in a ErrorUtils something so we could re-use it ?